### PR TITLE
command: update edit-validation-sets to craft-cli

### PR DESCRIPTION
**Pull request by sergiusens**
This is a minimal code change migration from Snapcraft's snapcraft_legacy module to snapcraft. Adaptations to use craft-cli for output has been made, as well as updates to how file edit takes place by migrating it to use Path.

The tests have been updated to follow the newer conventions of the code base, but the gist of them has not changed at all.

Lastly, the only UI change is that validation sets have been separated into their own command group.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----
CRAFT-1402

### CODE REVIEW

**Comment by tigarmo** [comment]
It's a bit hard to tell what's code that has just been moved and what's code that's materially changed. Can you point at which changes we should pay special attention to?

File: snapcraft/cli.py [hunk 1]
```diff
@@ -101,18 +101,23 @@
...
-            commands.StoreLegacyGatedCommand,
-            commands.StoreLegacyListValidationSetsCommand,
             commands.StoreLegacyRegisterKeyCommand,
             commands.StoreLegacySignBuildCommand,
-            commands.StoreLegacyValidateCommand,
             commands.StoreLegacyListKeysCommand,
         ],
     ),
+    craft_cli.CommandGroup(
+        "Store Validation Sets",
```
Comment (sergiusens) [inline snapcraft/cli.py:113]: I moved the commands to a new command group

File: snapcraft/commands/validation_sets.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            sequence=parsed_args.sequence,
+            revision=revision,
+            snaps=snaps,
+        )
+
+        with emit.pause():
+            edited_validation_sets = _edit_validation_sets(unverified_validation_sets)
+
+        if edited_validation_sets == yaml.safe_load(unverified_validation_sets):
+            emit.message("No changes made")
```
Comment (sergiusens) [inline snapcraft/commands/validation_sets.py:123]: click's echo was replaced with emit

File: snapcraft/commands/validation_sets.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                signed_validation_sets=signed_validation_sets
+            )
+
+
+def _edit_validation_sets(validation_sets: str) -> Dict[str, Any]:
+    """Spawn an editor to modify the validation-sets."""
+    editor_cmd = os.getenv("EDITOR", "vi")
+
+    with tempfile.NamedTemporaryFile() as temp_file:
+        file_template_path = Path(temp_file.name)
```
Comment (sergiusens) [inline snapcraft/commands/validation_sets.py:141]: the context management here has been simplified thanks to use of Path and its write_text and read_text methods

**Comment by sergiusens** [comment]
@tigarmo I added the relevant comments, the changes are really simple in themselves and out of necessity as "click" needs to be replaced with emit; and pydantic required changes in the edit method so I moved it to Path which simplified it

**Comment by tigarmo** [comment]
thanks, that helped a lot!

**Review by tigarmo — APPROVED** [review]

**Pull request merged**